src/wp-includes/option.php

```diff
@@ -1093,16 +1103,15 @@ function delete_option( $option ) {
 	$result = $wpdb->delete( $wpdb->options, array( 'option_name' => $option ) );
 
 	if ( ! wp_installing() ) {
-		if ( 'yes' === $row->autoload ) {
+		if ( 'yes' === $row->autoload || 'on' === $row->autoload || 'auto-on' === $row->autoload ) {
```

Comment:
Keep in mind that `auto` will also be autoloaded for now.
```suggestion
		if ( 'yes' === $row->autoload || 'on' === $row->autoload || 'auto-on' === $row->autoload || 'auto' === $row->autoload ) {
```